### ISSUE TITLE
AudioPlayer.get_pos() always is 0

### ISSUE THREAD

**Issue by toby3d** [godotengine/godot#8438]
**Operating system or device - Godot version:** Linux Mint Mate 18.1 (x86_64), last Godot Engine 3.0 build 
**Issue description:** `AudioPlayer.get_pos()` always returns 0, even if the music is playing right now.
**Steps to reproduce:** Add root Node2D, add AudioPlayer with some ogg music, add script:
```
func _ready():
	$AudioPlayer.play()
	set_process(true)

func _process(delta):
	print(str($AudioPlayer.get_pos()))
```
Music plays. Information about the current position of the music being played is not updated.

**Comment by attilix** [comment]
It's not a bug. The get_pos() method is not programmed it always returns a constant value of 0

**Comment by toby3d** [comment]
@attilix Then why implement something that does not work? For example, in Input has functional like _just_pressed(), but it is works.

**Comment by bojidar-bg** [comment]
@attilix It is a bug since it doesn't work as expected.
If @toby3d were requesting additional functionality (hint: he doesn't) it would have been an enhancement or a feature request, but since the current advertised functionality is not working, it's a bug.

**Comment by bojidar-bg** [comment]
Also, it IS programmed already:
* [`modules/opus/audio_stream_opus.cpp:343`](https://github.com/godotengine/godot/blob/df61dc4b2bd54a5a40c515493c76f5a458e5b541/modules/opus/audio_stream_opus.cpp#L343)
* [`scene/resources/audio_stream_sample.cpp:66`](https://github.com/godotengine/godot/blob/df61dc4b2bd54a5a40c515493c76f5a458e5b541/scene/resources/audio_stream_sample.cpp#L66)
* [`modules/stb_vorbis/audio_stream_ogg_vorbis.cpp:89`](https://github.com/godotengine/godot/blob/df61dc4b2bd54a5a40c515493c76f5a458e5b541/modules/stb_vorbis/audio_stream_ogg_vorbis.cpp#L89)

You shouldn't get confused by [`servers/audio/audio_stream.h:47`](https://github.com/godotengine/godot/blob/df61dc4b2bd54a5a40c515493c76f5a458e5b541/servers/audio/audio_stream.h#L47):
```
virtual float get_pos() const = 0;
```
This `= 0;` is the C++ marker for "no implementation available", aka "abstract".

**Comment by toby3d** [comment]
@bojidar-bg I'll try to build the last commits in the master and test the function again. The example that I brought above is a the correct way to use?

**Comment by bojidar-bg** [comment]
@toby3d Not sure, looks legit to me though.

**Comment by attilix** [comment]
@bojidar-bg  Ups, You`re right, I was confused by that line, sorry for the incorrect statement @toby3d

**Comment by NNesh** [comment]
AudioPlayer.play() with argument also isn't work correctly.

**Comment by NNesh** [comment]
Fixed with #8522
